utils/post.js

```diff
@@ -236,6 +236,38 @@ const insertTopics = async (topics = []) => {
   }
 };
 
+/**
+ *
+ * @param {String} topics
+ */
+const insertSingleTopic = async (topic) => {
+  const lastTopic = await Topics.findOne({
+    order: [['topic_id', 'DESC']],
+    limit: 1,
+    raw: true
+  });
+
+  const topicIndex = parseInt(lastTopic.topic_id, 10) + parseInt(1, 10);
+
+  try {
+    await Topics.findOrCreate({
+      where: {name: topic},
+      defaults: {
+        topic_id: topicIndex,
+        name: topic,
+        icon_path: '',
+        is_custom_topic: true,
+        created_at: new Date(),
+        categories: ''
+      }
+    });
+
+    return topicIndex;
+  } catch (e) {
+    console.log(e);
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `insertSingleTopic` function has a potential race condition. If two instances of this function run concurrently, they might both fetch the same last topic ID and try to insert a new topic with the same incremented ID. This could lead to a unique constraint violation if `topic_id` is a unique field in your database.

To avoid this, you should let the database handle the generation of unique IDs. Most databases have auto-increment functionality for this purpose. If you're using Sequelize (which it looks like you are), you can set `autoIncrement: true` on the `topic_id` field definition.

Also, error handling in this function could be improved. Currently, if an error occurs during the `findOrCreate` operation, it's logged to the console but not propagated further. This means that any function calling `insertSingleTopic` won't know if the operation was successful or not. It would be better to throw the error after logging it, so that it can be handled appropriately by the caller.

```diff
const insertSingleTopic = async (topic) => {
  try {
    const [newTopic, created] = await Topics.findOrCreate({
      where: {name: topic},
      defaults: {
        name: topic,
        icon_path: '',
        is_custom_topic: true,
        created_at: new Date(),
        categories: ''
      }
    });

    return newTopic.topic_id;
  } catch (e) {
    console.log(e);
+    throw e;
  }
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->